cost/google/idle_compute_instances/google_idle_compute_instances.pt

```diff
@@ -209,77 +216,102 @@ script "js_calculated_utilization", type: "javascript" do
+    for (var j = 0; j < ds_instances[i].instances.length; j++) {
+      collected_instances.push(ds_instances[i].instances[j])
+    }
+  }
+  EOS
+end
+
 script "js_add_instance_data", type: "javascript" do
-  result "request"
-  parameters "project","zone", "instance_id"
+  result "instances_with_data"
+  parameters "ds_collected_instances", "ds_calculated_utilization"
   code <<-EOS
-  request = {
-    "auth": "auth_google",
-    "host": "www.googleapis.com",
-    "verb": "GET",
-    "path": "/compute/v1/projects/"+project+"/zones/"+zone+"/instances",
-    "headers": {
-      "User-Agent": "RS Policies",
-      "Content-Type": "application/json"
-    },
-    "query_params": {
-      "filter": "(id="+instance_id+") AND (status=RUNNING)"
+  var instances_with_data = []
+  for (var i = 0; i < ds_calculated_utilization.length; i++) {
+    var instance_id = ds_calculated_utilization[i].instance_id
+    var data_record = _.find(ds_collected_instances, function(record) { return record.id === instance_id })
+    if (data_record === null || data_record === undefined) {
+      instances_with_data.push({
+        id: instance_id,
+        hostname: "",
+        selfLink: "",
+        label_instance_name: ds_calculated_utilization[i].label_instance_name,
+        status: "",
+        zone: ds_calculated_utilization[i].zone,
+        labels: {},
+        cpu_average: parseFloat(ds_calculated_utilization[i].cpu_average).toFixed(2),
+        cpu_maximum: parseFloat(ds_calculated_utilization[i].cpu_maximum).toFixed(2),
+        cpu_minimum: parseFloat(ds_calculated_utilization[i].cpu_minimum).toFixed(2),
+        mem_average: parseFloat(ds_calculated_utilization[i].mem_average).toFixed(2),
+        mem_maximum: parseFloat(ds_calculated_utilization[i].mem_maximum).toFixed(2),
+        mem_minimum: parseFloat(ds_calculated_utilization[i].mem_minimum).toFixed(2),
+        projectId: ds_calculated_utilization[i].projectId
+      })
+    } else {
+      instances_with_data.push({
+        id: instance_id,
+        hostname: data_record.name || data_record.id,
+        selfLink: data_record.selfLink,
+        label_instance_name: ds_calculated_utilization[i].label_instance_name,
+        status: data_record.status,
+        zone: ds_calculated_utilization[i].zone,
+        labels: data_record.labels,
+        cpu_average: parseFloat(ds_calculated_utilization[i].cpu_average).toFixed(2),
+        cpu_maximum: parseFloat(ds_calculated_utilization[i].cpu_maximum).toFixed(2),
+        cpu_minimum: parseFloat(ds_calculated_utilization[i].cpu_minimum).toFixed(2),
+        mem_average: parseFloat(ds_calculated_utilization[i].mem_average).toFixed(2),
+        mem_maximum: parseFloat(ds_calculated_utilization[i].mem_maximum).toFixed(2),
+        mem_minimum: parseFloat(ds_calculated_utilization[i].mem_minimum).toFixed(2),
+        projectId: ds_calculated_utilization[i].projectId
+      })
     }
-    "ignore_status": [403,404]
```

Comment:
@Rangelmv23 hey!

Before we used `ds_add_instance_data` to combine instance data and utilization, `ds_instance_data` used a GCP endpoint to retrieve data for each instance, 1 request per instance, this was pretty slow, so now we changed this, and instead we are using the new `ds_instances` to retrieve all the instance's data and then combine it in a single request.

The short answer is that everything got changed, so that line was not necessary anymore.